sources/packages/api/src/auth/guards/sin-validation.guard.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable, CanActivate, ExecutionContext } from "@nestjs/common";
+import { Reflector } from "@nestjs/core";
+import { StudentService } from "../../services";
+import { CHECK_SIN_VALIDATION_KEY } from "../decorators/check-sin-status.decorator";
+import { IUserToken } from "../userToken.interface";
+/**
+ * This guard validates an API for Valid Sin if it is decorated with @CheckRestriction.
+ */
+@Injectable()
+export class SinValidationGuard implements CanActivate {
```

Comment:
Can you please change the class name to SINValidationGuard from SinValidationGuard.